### PR TITLE
[lambda] update context extraction

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Updates how context extraction is done.

### Motivation
Context might not be the second, but the third argument. Which our patch might crash AWS Lambda function if this happens.

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/lambda/handler.js [hunk 1]
```diff
@@ -59,14 +59,31 @@ function crashFlush () {
...
+ *
+ * @param {*} args any amount of arguments
+ * @returns the context, if extraction was succesful.
+ */
+function extractContext (args) {
+  let context = args.length > 1 ? args[1] : undefined
+  if (context === undefined || context.getRemainingTimeInMillis === undefined) {
+    context = args.length > 2 ? args[2] : undefined
+    if (context === undefined || context.getRemainingTimeInMillis === undefined) {
+      throw Error('Could not extract context')
```
Comment (astuyve) [inline packages/dd-trace/src/lambda/handler.js:73]: The tracer shouldn't crash in this case, is there a try/catch block for this?

**Review by astuyve — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/lambda/handler.js (hunk 1 above)
Comment (duncanista) [reply packages/dd-trace/src/lambda/handler.js:73]: Yup, I was figuring out the best place, currently it's on our instrumentation, couldn't find a better place yet!

**Review by Qard — DISMISSED** [review]

**Comment by duncanista** [comment]
log on cold start is currently not working - will research tomorrow how to honor global variable.

File: packages/dd-trace/src/lambda/handler.js [hunk 2]
```diff
@@ -50,7 +55,9 @@ function crashFlush () {
       [ERROR_TYPE]: error.name
     })
   } else {
-    log.warn('An impending timeout was reached, but no root span was found. No error will be tagged.')
+    if (__isColdStart) {
+      log.warn('An impending timeout was reached, but no root span was found. No error will be tagged.')
```
Comment (astuyve) [inline packages/dd-trace/src/lambda/handler.js:59]: If we're having issues with this cold start feature, it's really a nice-to-have. I'd suggest removing it and making this a `log.debug` and we move forward.

**Review by astuyve — APPROVED** [review]

**Review by astuyve — APPROVED** [review]

**Pull request merged**